node/src/client/mod.rs

```diff
@@ -39,17 +39,27 @@ use snarkvm::{
         store::ConsensusStorage,
         Ledger,
     },
+    prelude::block::Transaction,
 };
 
 use anyhow::Result;
 use core::future::Future;
+use lru::LruCache;
 use parking_lot::Mutex;
 use std::{
     net::SocketAddr,
-    sync::{atomic::AtomicBool, Arc},
+    num::NonZeroUsize,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize, Ordering::Relaxed},
+        Arc,
+    },
 };
 use tokio::task::JoinHandle;
 
+const VERIFICATION_CONCURRENCY_LIMIT: usize = 6; // 8 deployments of MAX_NUM_CONSTRAINTS will run out of memory.
+
+type TransactionContents<N> = (SocketAddr, UnconfirmedTransaction<N>, Transaction<N>);
```

Comment:
we might want to comment on why we're preserving the serialized transaction too (i.e. for faster propagation)

Reply:
989e741ee